lib/spaceship/client.rb

```diff
@@ -184,18 +196,36 @@ def login(user = nil, password = nil)
 
     def with_retry(tries = 5, &block)
       return block.call
-    rescue Faraday::Error::TimeoutError, AppleTimeoutError => ex # New Faraday version: Faraday::TimeoutError => ex
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError, AppleTimeoutError => ex # New Faraday version: Faraday::TimeoutError => ex
       unless (tries -= 1).zero?
         logger.warn("Timeout received: '#{ex.message}'.  Retrying after 3 seconds (remaining: #{tries})...")
         sleep 3
         retry
       end
-
+      raise ex # re-raise the exception
+    rescue UnauthorizedAccessError => ex
+      if @loggedin
+        msg = "Auth error received: '#{ex.message}'. Login in again then retrying after 3 seconds (remaining: #{tries})..."
+        puts msg
+        logger.warn msg
```

Comment:
I agree regarding verbosity, however I think `spaceship` almost never prints out stuff, except when selecting the team. I think we should keep it this way cc @snatchev